lib/editor/components/timetable/TripSeriesModal.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+// $FlowFixMe: Flow doesn't know about useState: https://stackoverflow.com/questions/53105954/cannot-import-usestate-because-there-is-no-usestate-export-in-react-flow-with
+import React, {useState} from 'react'
+import {Button, Modal} from 'react-bootstrap'
+
+import * as tripActions from '../../actions/trip'
+import {getComponentMessages} from '../../../common/util/config'
+import HourMinuteInput from '../HourMinuteInput'
+import type {Trip} from '../../../types'
+
+type Props = {
+  addNewTrip: typeof tripActions.addNewTrip,
+  constructNewRow: (trip: ?Trip, tripSeriesStartTime: number) => ?Trip,
+  onClose: () => void,
+  show: boolean
+}
+
+const TripSeriesModal = (props: Props) => {
+  const [startTime, setStartTime] = useState(0)
+  const [headway, setHeadway] = useState(0)
+  const [endTime, setEndTime] = useState(0)
+
+  const messages = getComponentMessages('TripSeriesModal')
+
+  const onClickGenerate = () => {
+    const {addNewTrip, constructNewRow, onClose} = props
+    for (let time = startTime; time <= endTime; time += headway) {
+      addNewTrip(constructNewRow(null, time))
+    }
+    setStartTime(0)
+    setEndTime(0)
+    setHeadway(0)
+    onClose()
+  }
+
+  const {Body, Footer, Header, Title} = Modal
+  const {onClose, show} = props
+  return (
+    <Modal show={show} onHide={onClose}>
+      <Header><Title>{messages('createTripSeriesQuestion')}</Title></Header>
+      <Body>
+        {messages('createTripSeriesBody')}
+        <br />
+        {messages('startTime')} <div><HourMinuteInput onChange={setStartTime} seconds={startTime} standaloneInput /></div>
+        {messages('headway')} <div><HourMinuteInput onChange={setHeadway} seconds={headway} standaloneInput /></div>
+        {messages('endTime')} <div><HourMinuteInput onChange={setEndTime} seconds={endTime} standaloneInput /></div>
+      </Body>
+      <Footer>
+        <Button
+          bsStyle='primary'
+          disabled={!startTime || !endTime || !headway}
+          onClick={onClickGenerate}
```

Comment:
If you set the start time to be after the end time it just silently does nothing which I guess is fine? but it might be better to catch the error and then set the input border to red to indicate there's a problem?

Reply:
Added in https://github.com/ibi-group/datatools-ui/pull/892/commits/9616d30f18c1ad70ea52de68cae60a87770db3f9